LeaderboardBackend.Test/Features/Users/AccountConfirmationTests.cs

```diff
@@ -0,0 +1,148 @@
+    {
+        s_factory.ResetDatabase();
+        _scope.Dispose();
+    }
+
+    [Test]
+    public async Task ResendConfirmation_Unauthorised()
+    {
+
+        HttpResponseMessage res = await Client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+        res.Should().HaveStatusCode(HttpStatusCode.Unauthorized);
+    }
+
+    [Test]
+    public async Task ResendConfirmation_NotFound_ShouldGet401()
+    {
+        string token = _authService.GenerateJSONWebToken(new()
+        {
+            Email = "unknown@user.com",
+            Password = "password",
+            Username = "username",
+        });
+
+        Client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await Client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+
+        res.Should().HaveStatusCode(HttpStatusCode.Unauthorized);
+    }
+
+    [Test]
+    public async Task ResendConfirmation_Conflict()
+    {
+        // TODO: Call UserService instead, once we're able to set a user's role with it.
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        User user = new()
+        {
+            Email = "email",
+            Password = "password",
+            Username = "username",
+            Role = UserRole.Confirmed,
+        };
+        context.Add<User>(user);
+        context.SaveChanges();
+        string token = _authService.GenerateJSONWebToken(user);
+
+        Client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await Client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+
+        res.Should().HaveStatusCode(HttpStatusCode.Conflict);
+    }
+
+    [Test]
+    public async Task ResendConfirmation_EmailFailedToSend()
+    {
+        Mock<IEmailSender> emailSenderMock = new();
+        emailSenderMock.Setup(e =>
+            e.EnqueueEmailAsync(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())
+        ).Throws(new System.Exception());
+        HttpClient client = s_factory.WithWebHostBuilder(builder =>
+        {
+            builder.ConfigureTestServices(services =>
+            {
+                services.AddScoped<IEmailSender>(_ => emailSenderMock.Object);
+            });
+        })
+        .CreateClient();
+
+        IUserService userService = _scope.ServiceProvider.GetRequiredService<IUserService>();
+        CreateUserResult result = await userService.CreateUser(new()
+        {
+            Email = "email",
+            Password = "password",
+            Username = "username",
+        });
+        string token = _authService.GenerateJSONWebToken(result.AsT0);
+
+        client.DefaultRequestHeaders.Authorization = AuthenticationHeaderValue.Parse($"Bearer {token}");
+        HttpResponseMessage res = await client.PostAsync(Routes.RESEND_CONFIRMATION, null);
+        res.Should().HaveStatusCode(HttpStatusCode.InternalServerError);
+    }
```

Comment:
I'm not sure if we need to test something that shouldn't actually happen. We should probably hold off on this test case until we figure out what behavior we actually want in this scenario.

Reply:
I'd argue instead that it's worth making sure we have the behaviour we've currently got. When we change the action's behaviour later on, then we change the test.